pkg/pillar/cmd/zedrouter/appcontainer.go

```diff
@@ -66,23 +70,33 @@ func appStatsAndLogCollect(ctx *zedrouterContext) {
 				return
 			}
 
+			var acNum int
 			collectTime := time.Now() // all apps collection assign the same timestamp
-			log.Infof("appStatsAndLogCollect: containerStats, timer loop")
 			for _, st := range items {
 				status := st.(types.AppNetworkStatus)
 				if status.GetStatsIPAddr != nil {
-					acMetrics, err := appContainerGetStats(status.GetStatsIPAddr)
+					// get a list of containers and client handle
+					cli, containers, err := getAppContainers(status)
 					if err != nil {
-						log.Errorf("appStatsAndLogCollect: can't get App %s Container Metrics on %s, %v",
+						log.Errorf("appStatsAndLogCollect: can't get App Containers %s on %s, %v",
 							status.UUIDandVersion.UUID.String(), status.GetStatsIPAddr.String(), err)
 						continue
 					}
-					acMetrics.UUIDandVersion = status.UUIDandVersion
-					acMetrics.CollectTime = collectTime
-					ctx.pubAppContainerMetrics.Publish(acMetrics.Key(), acMetrics)
-					getAppContainerLogs(status, lastLogTime)
+					acNum += len(containers)
+
+					// collect container stats, and publish to zedclient
+					acMetrics := getAppContainerStats(status, cli, containers)
+					if len(acMetrics.StatsList) > 0 {
+						acMetrics.UUIDandVersion = status.UUIDandVersion
+						acMetrics.CollectTime = collectTime
+						ctx.pubAppContainerMetrics.Publish(acMetrics.Key(), acMetrics)
+					}
+
+					// collect container logs and send through the logging system
+					getAppContainerLogs(status, lastLogTime, cli, containers)
 				}
 			}
+			log.Infof("appStatsAndLogCollect: containerStats, %d processed. reset timer", acNum)
```

Comment:
Will this be emitted periodically? If so a log.Debugf would make more sense to avoid log bloat.

Reply:
it's every 5 minutes by default. we can make a comment saying converting this to debug if noisy, but currently it's the only thing emitted from this go routine and it's good to know it's running or not.

Reply:
That is better handled by introducing another StillRunning .touch file. But we can do that later when these log messages become a problem.

Reply:
Sure. so far i have noticed is that, the App side containers can crash or not up running, and we don't get anything. at least this log indicate we have got zero number of containers in the reply, and the this goroutine is running as expected.